### PR TITLE
Bump Airflow to 2.4.0, standardize version bump process

### LINKED ISSUE #732: Upgrade to Airflow 2.4.0
## Description

[Airflow 2.4.0](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-4-0-2022-09-19) has some exciting new features and a number of deprecations as well. We should upgrade the version and adjust the code to conform to the new specifications.

## Additional context

This was going to be done as part of #731 but it proved a greater effort than what time would allow.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #732 by @AetherUnbound

This also addresses some security issues that dependabot identified.

## Description

This PR updates our Airflow version to 2.4.0. I've also borrowed the pattern set up in #656 and applied it to the Airflow version, so we only have to update it in one place now!

As part of the version update, I've also updated the [`schedule_interval` parameter to be `schedule` instead](https://airflow.apache.org/docs/apache-airflow/2.4.0/release_notes.html#deprecation-of-schedule-interval-and-timetable-arguments-25410). What's unfortunate however is that the `airflow.models.dag.Dag` model _does not_ have a `schedule` attribute, you still must use `schedule_interval` when referencing it on the DAG object. So there are a few instances where we were not able to update that value.

I added a bunch of deprecation warning filters for our tests to make the warning summary more relevant (presently there should be no warnings). Most of the warnings were either 1) irrelevant & unfixable or 2) due to upstream dependencies.

Lastly, and probably most notably:

> ⚠️ **Developer note** ⚠️
> The extra argument for `AIRFLOW_CONN_AWS_DEFAULT` has been updated from `host` to `endpoint_url`:
> _Before_: `AIRFLOW_CONN_AWS_DEFAULT=aws://test_key:test_secret@?region_name=us-east-1&host=http://s3:5000`
> _After_: `AIRFLOW_CONN_AWS_DEFAULT=aws://test_key:test_secret@?region_name=us-east-1&endpoint_url=http://s3:5000`
>
> You will continue to receive warnings about `extra["host"]` unless this value is changed in your `.env` file!

## Testing Instructions

1. `just build`
1. `just test --extended` (and ensure there are no warnings in the warnings summary)
1. Run a few DAGs to make sure everything functions as expected
1. `just airflow-version` should report `2.4.0`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/providers/provider_dag_factory.py [hunk 1]
```diff
@@ -92,7 +92,7 @@
 _DATE_RANGE_INNER_TEMPLATE = "macros.ds_add(ds, -{} )"
 DATE_RANGE_ARG_TEMPLATE = "{{{{" + _DATE_RANGE_INNER_TEMPLATE + "}}}}"
 DATE_PARTITION_ARG_TEMPLATE = Template(
-    "$media_type/$provider_name/{{ date_partition_for_prefix(dag.schedule, dag_run.logical_date, $reingestion_date ) }}"  # noqa
+    "$media_type/$provider_name/{{ date_partition_for_prefix(dag.schedule_interval, dag_run.logical_date, $reingestion_date ) }}"  # noqa
```
Comment (rwidom) [inline openverse_catalog/dags/providers/provider_dag_factory.py:95]: This is so odd! I thought maybe `dag.params['schedule']` might work, but no:

```ERROR - Failed to execute task: 'airflow.models.param.ParamsDict object' has no attribute 'schedule'.```

**Review by rwidom — APPROVED** [review]
I followed the testing instructions and ran phylopic, and all went well.

File: requirements_prod.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # PYTHON=3.10
-apache-airflow[amazon,postgres,http]==2.3.4
+apache-airflow[amazon,postgres,http]==2.4.0
```
Comment (obulat) [inline requirements_prod.txt:2]:
```suggestion
apache-airflow[amazon,postgres,http]==2.4.0b1
```

Comment (obulat) [reply requirements_prod.txt:2]: This is a patched version. But I'm not sure if this should apply to amazon, postgres and http packages, too.

**Review by krysal — APPROVED** [review]
Built and ran the legacy smk script and everything looks good.

**Review by stacimc — APPROVED** [review]
This looks good to me. I tested with Cleveland, Jamendo, and Wikimedia Reingestion. I tried to test the audio data refresh but am unable to get it working due to environment problems; if I get those sorted I'll test that as well, but I see no reason it shouldn't work 🙂

The additions to ignore warnings are very nice (and individually well documented)! I got no warnings, after I updated my local `.env`. Shame about `dag.schedule_interval`.

Edit: forgot to add that the only other instances of `schedule_interval` I found were in several DAGs in the `retired` dir. They're retired so we certainly don't _need_ to change them. What do you think @AetherUnbound?

File: openverse_catalog/dags/providers/provider_dag_factory.py (hunk 1 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/providers/provider_dag_factory.py:95]: It definitely made me do a double take at first 😞

File: requirements_prod.txt (hunk 2 above)
Comment (AetherUnbound) [reply requirements_prod.txt:2]: I believe 2.4.0b1 is the _first version in which the fix was introduced_, but it's not the most recent version. Looking at the commits for both, you can see that [v2.4.0b1](https://github.com/apache/airflow/commits/2.4.0b1) was published before [v2.4.0](https://github.com/apache/airflow/commits/2.4.0) (IIRC semver dictates that `X.Y.Zaaa` is a pre-release version where is `X.Y.Z` is the proper release). So using 2.4.0 should be good!

File: tests/utilities/dag_doc_gen/test_dag_doc_generation.py [hunk 3]
```diff
@@ -42,14 +42,12 @@ class DagMock(NamedTuple):
 )
 @pytest.mark.parametrize("provider_workflow", ["foo_bar", None])
 def test_get_dags_info(
-    schedule_interval, doc, expected_doc, catchup, tags, type_, provider_workflow
+    schedule, doc, expected_doc, catchup, tags, type_, provider_workflow
 ):
-    dag = DagMock(
-        schedule_interval=schedule_interval, doc_md=doc, catchup=catchup, tags=tags
-    )
+    dag = DagMock(schedule_interval=schedule, doc_md=doc, catchup=catchup, tags=tags)
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply tests/utilities/dag_doc_gen/test_dag_doc_generation.py:47]: Good call, can do!

**Comment by AetherUnbound** [comment]
Ah, great thought on the retired DAGs. I'll revert those changes, we definitely don't need to change them. I used too aggressive a find-and-replace 😅

**Pull request merged**